File: 9-PalindromeNumber/9-PalindromeNumber.cpp
// Last updated: 4/24/2026, 8:15:31 AM
class Solution {
public:
    bool isPalindrome(int x) {
        int xx = x;
        long revNum = 0L;
        while (xx > 0) {
            int digit = xx % 10;
            revNum = revNum * 10L + digit;
            xx /= 10;
        }
        return revNum == x;
    }
};

File: 9-PalindromeNumber/9-PalindromeNumber.java
// Last updated: 4/24/2026, 8:15:28 AM
class Solution {
    public boolean isPalindrome(int x) {
        // String input = String.valueOf(x);
        // return input.equals(new StringBuilder(input).reverse().toString());

        if (x < 0)
            return false;
        int xx = x;
        int revNum = 0;
        while (x > 0) {
            int lastDigit = x % 10;
            revNum = revNum * 10 + lastDigit;
            x /= 10;
        }
        return xx == revNum;
    }
}